mountinfo/mounted_linux.go

```diff
@@ -34,24 +49,50 @@ func mountedByOpenat2(path string) (bool, error) {
 	return false, &os.PathError{Op: "openat2", Path: path, Err: err}
 }
 
-func mounted(path string) (bool, error) {
-	path, err := normalizePath(path)
+// mountedFast is similar to MountedFast, plus it returns the normalized path.
+func mountedFast(inPath string) (path string, mounted, sure bool, err error) {
+	// Shortcut: root is always mounted.
+	if path == string(os.PathSeparator) {
+		mounted = true
+		sure = true
+		return
+	}
+
+	path, err = normalizePath(inPath)
 	if err != nil {
-		return false, err
+		return
+	}
+
+	// Check for root again, in case it was not normalized.
+	if path == string(os.PathSeparator) {
+		mounted = true
+		sure = true
+		return
 	}
+
 	// Try a fast path, using openat2() with RESOLVE_NO_XDEV.
-	mounted, err := mountedByOpenat2(path)
+	mounted, err = mountedByOpenat2(path)
 	if err == nil {
-		return mounted, nil
+		sure = true
+		return
 	}
 	// Another fast path: compare st.st_dev fields.
 	mounted, err = mountedByStat(path)
 	// This does not work for bind mounts, so false negative
 	// is possible, therefore only trust if return is true.
 	if mounted && err == nil {
-		return mounted, nil
+		sure = true
+	}
+
+	return
+}
+
+func mounted(path string) (bool, error) {
+	path, mounted, sure, err := mountedFast(path)
+	if sure {
+		return mounted, err
 	}
```

Comment:
I slightly prefer having this check as (even when we have tests) as an extra check.

```
if sure && err == nil 
```

Reply:
Sure.

Also, I just realized that if `normalizePath` returns an error, it does not make sense to continue with parsing mountinfo, so this code needs to be changed anyway.

Will fix.